### PR TITLE
Limit Network text view to 2^16 characters to prevent it from crashing

### PR DESCRIPTION
**Pull request by CoderDake**
For images, the request and response bodies can get large which cause the Text view to hang DevTools.

I've hardcoded the 2^16 character limit for now, but I could be persuaded to allow an override in case other textviews have different needs.

Fixes https://github.com/flutter/devtools/issues/5973 

### Truncated
Screenshot 2023-08-25 at 3 47 54 PM
### Under Character Limit

Screenshot 2023-08-25 at 3 47 58 PM

### CODE REVIEW

**Comment by bkonyi** [comment]
I'm fine with this for now, but I think this warrants further investigation as to why it causes a hang. We shouldn't have an issue rendering 2^16 characters. Can you file an issue to follow up on this?

File: packages/devtools_app/lib/src/shared/common_widgets.dart [hunk 1]
```diff
@@ -1264,6 +1264,30 @@ class JsonViewer extends StatefulWidget {
   State<JsonViewer> createState() => _JsonViewerState();
 }
 
+class TextViewer extends StatelessWidget {
+  const TextViewer({
+    super.key,
+    required this.text,
+  });
+  final String text;
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/common_widgets.dart:1272]: This needs some whitespace.

File: packages/devtools_app/lib/src/shared/common_widgets.dart [hunk 2]
```diff
@@ -1264,6 +1264,30 @@ class JsonViewer extends StatefulWidget {
   State<JsonViewer> createState() => _JsonViewerState();
 }
 
+class TextViewer extends StatelessWidget {
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/common_widgets.dart:1267]: This needs some documentation. `TextViewer` is very generic without context.

File: packages/devtools_app/lib/src/shared/common_widgets.dart [hunk 3]
```diff
@@ -1264,6 +1264,30 @@ class JsonViewer extends StatefulWidget {
...
+    super.key,
+    required this.text,
+  });
+  final String text;
+  static const _maxLength = 65536; //2^16
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final String displayText;
+    // Limit the length of the displayed text to _MAX_LENGTH
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/common_widgets.dart:1278]: We should reference the issue here.

Comment (CoderDake) [reply packages/devtools_app/lib/src/shared/common_widgets.dart:1278]: I'll reference it in the network code where we reference the TextViews.

File: packages/devtools_app/lib/src/shared/common_widgets.dart [hunk 4]
```diff
@@ -1264,6 +1264,36 @@ class JsonViewer extends StatefulWidget {
...
+  });
+
+  final String text;
+  final int maxLength;
+  final TextStyle? style;
+
+  @override
+  Widget build(BuildContext context) {
+    final String displayText;
+    // Limit the length of the displayed text to _MAX_LENGTH
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/common_widgets.dart:1284]: s/`_MAX_LENGTH`/`maxLength`/

File: packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart [hunk 5]
```diff
@@ -356,8 +358,10 @@ class HttpTextResponseViewer extends StatelessWidget {
 
         return switch (currentLocalResponseType) {
           NetworkResponseViewType.json => JsonViewer(encodedJson: responseBody),
-          NetworkResponseViewType.text => Text(
-              responseBody,
+          // Using TextViewer fixes
+          // https://github.com/flutter/devtools/issues/5973
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart:362]: The above issue should be closed when this PR lands, but I'd file a new issue for this as it's a bandaid fix if we don't know why we're seeing the hang in the first place.

Comment (CoderDake) [reply packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart:362]: Created the following issues in response to this discussion:
- https://github.com/flutter/devtools/issues/6261
- https://github.com/flutter/devtools/issues/6263

**Review by bkonyi — APPROVED** [review]

**Pull request merged**